File: MVC_DiceGame/MVC_DiceGame_java/view/Console.java
package view;

public class Console implements IView {
  public int getInputChar() {
    try {
      int c = System.in.read();
      while (c == '\r' || c =='\n') {
        c = System.in.read();
      }
      return c;
    } catch (java.io.IOException e) {
      System.out.println("" + e);
      return 0;
    }
  }
  
  public boolean wantsToPlay() {
    return getInputChar() != 'q';
  }
  
  public void DisplayResult(boolean a_win, int a_dice1Val, int a_dice2Val) {
    if (a_win) {
      System.out.println("Win: " + a_dice1Val + " " + a_dice2Val);
    } else {
      System.out.println("Lose:" +a_dice1Val + " " + a_dice2Val);
    }
  }
  
  public void DisplayWelcomeMessage() {
    System.out.println("Hello Dice Game World");
  }
  
  public void DisplayInstructions() {
    System.out.println("Any key to play 'q' to quit");
  }
}

File: MVC_DiceGame/MVC_DiceGame_Swift_3.0.1/MVC_DiceGame_Swift_3.0.1/view/Console.swift
//
//  Console.swift
//  MVC_DiceGame_Swift_3.0.1
//
//  Created by Sarpreet Singh on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

public class Console : IView {
    
    func getInputChar() -> String {
        var input : String = readLine()!;
        while input == "\n" || input == "\r" || input.characters.count > 1{
            input = readLine()!;
        }
        return input;
    }
    
    func wantsToPlay() -> Bool {
        return getInputChar() != "q";
    }
    
    func displayResult(a_win : Bool, a_dice1Val : Int, a_dice2Val : Int) {
        if a_win {
            print("Win: " + String (a_dice1Val) + "  " + String (a_dice2Val));
        } else {
            print("Lose: " + String (a_dice1Val) + "  " + String (a_dice2Val));
        }
    }
    
    func displayWelcomeMessage() {
        print("Hello Dice Game World");
    }
    
    func displayInstructions() {
        print("Any key to play 'q' to quit");
    }
}
